test/libsyncengine/propagation/executor/executorworker/testexecutorworker.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "syncpal/syncpal.h"
+#include "propagation/executor/executorworker.h"
+#include "testincludes.h"
+
+
+namespace KDC {
+
+class TestExecutorWorker;
+
+using testFctPtr = void (TestExecutorWorker::*)();
+class TestExecutorWorker : public CppUnit::TestFixture {
+        CPPUNIT_TEST_SUITE(TestExecutorWorker);
+        CPPUNIT_TEST(testAffectedUpdateTree);
+        CPPUNIT_TEST(testTargetUpdateTree);
+        CPPUNIT_TEST_SUITE_END();
+
+    public:
+        void setUp() override;
+        void tearDown() override;
+
+    private:
+        void testAffectedUpdateTree();
+        void testTargetUpdateTree();
+
+        log4cplus::Logger _logger;
+        std::shared_ptr <ExecutorWorker> _testObj;
```

Comment:
Since there is already an ExecutorWorker in SyncPal, those 2 variables are probably not both useful. Either, use the `_executorWorker` attribute of SyncPal or, if possible, do not set up a SyncPal at all and only create an ExecutorWorker

Reply:
I'm trying to apply your suggestion, but i'm faccing the following issue:
The `_executorWorker` attribute of `SyncPal` is initialize only after the call to `SyncPal::Start()` wich need to set a user, drive and a syncro wich seems to be a little exagerated just for the `ExecutorWorker` tests. 

Only create an `ExecutorWorker` is not viable as the tested function rely at some point on the `_syncPal` memeber of `ExecutorWorker`.

Reply:
Ok, so perhaps this change should be the topic of a dedicated PR where we refactor deeper the way the tests are implemented.
We can resolve this thread